### PR TITLE
Support allocsize attribute

### PR DESCRIPTION
**Pull request by andybalholm**

### CODE REVIEW

File: ir/helper.go [hunk 1]
```diff
@@ -29,6 +29,21 @@ func (align AlignStack) String() string {
 	return fmt.Sprintf("alignstack(%d)", uint64(align))
 }
 
+// AllocSize is an attribute for functions like malloc.
+type AllocSize struct {
+	ElemSize uint64
+	N        uint64
+	HasN     bool
```
[the remark this answers is not in the document]
Comment (andybalholm) [reply ir/helper.go]: I think zero is a valid value for this field. It's not really the number of elements allocated; it's which parameter of the function contains the number allocated. And the parameters are numbered from zero. So we can't use zero for the sentinel.

We could potentially make the field signed, and use -1.

[the remark this answers is not in the document]
Comment (mewmew) [reply ir/helper.go]: > I think zero is a valid value for this field. It's not really the number of elements allocated; it's which parameter of the function contains the number allocated. And the parameters are numbered from zero. So we can't use zero for the sentinel.

Oh, I see. Thanks for the notice! I have not used `allocsize` before so had to refresh on the docs, you are indeed right.

In that case, I would suggest we rename the fields to `ElemSizeIndex` and `NElemsIndex`. What do you think?

> We could potentially make the field signed, and use -1.

I think it's a good idea to use `-1` for `NElemsIndex` so we only need to track one field. In that case, perhaps we should change the type from `uint64` to `int`. For consistency, lets use `int` for both `ElemSizeIndex` and `NElemsIndex`.

**Comment by mewmew** [comment]
Out of curiosity, did you update any of the test cases?

I'm just wondering why the testdata submodule had changes listed in this PR:

From https://github.com/llir/llvm/pull/58/files#diff-ef654c40ab4f1747fc699915d4f70902:

2018-12-29-000054_818x462_scrot

**Comment by andybalholm** [comment]
I didn't update the test cases. I have no idea why it switched to a different submodule version.

**Comment by mewmew** [comment]
Looks great as of commit fd2c6a9. Would you rebase on master and I'll merge this.

**Pull request merged**